torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -469,7 +473,22 @@ void RequestCallbackImpl::processRpc(
           [responseFuture,
            messageId,
            fromWorkerId,
-           weak = std::weak_ptr<FutureMessage>(wrappedRpcResponseFuture)]() {
+           weak = std::weak_ptr<FutureMessage>(wrappedRpcResponseFuture),
+           threadLocalState = ThreadLocalState(),
+           ctxId = autogradContext->contextId()]() {
```

Comment:
This is currently the only place on server-side where we have such a continuation by continuing processing with `addCallback`, correct?

Reply:
There are more in other types of requests. But IIUC, this is the only place we need to propagate context id. 

* In BACKWARD_AUTOGRAD_REQ, it does not need the context id when creating the `PropagateGradientsResp` obj
* In SCRIPT_CALL and SCRIPT_REMOTE_CALL, the context id is fixed by #36395.
* PYTHON_CALL and PYTHON_REMOTE_CALL currently do not yield. But when we add async user function, we also need to propagate the TLS here.
* SCRIPT_RREF_FETCH_CALL and PYTHON_RREF_FETCH_CALL do not need the context id either.